src/templating/content.md

```diff
@@ -0,0 +1,654 @@
+    name: proxy-container
+    ports:
+    - containerPort: 80
+  - env:
+    - name: ENV
+      value: production
+    image: k8s.gcr.io/busybox
+    name: test-container
+```
+
+The patch functionality works in a similar way of `yq`, but setup for kustomize is more verbose.
+
+_Should you stop using kustomize in favour or `yq`?_
+
+Kustomize is designed to map changes and resources easily.
+
+You should create another folder at the same level at the previous one:
+
+```terminal|command=1,2,3|title=bash
+cd ..
+mkdir dev
+cd dev
+```
+
+You can create another `kustomization.yaml`:
+
+```yaml|highlight=3-4|title=kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+bases:
+  - ../prod
+patchesStrategicMerge:
+  - pod-patch-env.yaml
+```
+
+The new configuration extends the base configuration `prod`, so there's no need to reapply all patches.
+
+Instead, you can create a single YAML file that changes the environment variable:
+
+```yaml|title=pod-patch-env.yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-pod
+spec:
+  containers:
+    - name: test-container
+      env:
+      - name: ENV
+        value: dev
+```
+
+If you run kustomize this time, the output will be different:
+
+```terminal|command=1|title=bash
+kubectl kustomize dev
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-pod
+spec:
+  containers:
+  - image: envoyproxy/envoy:v1.12.2
+    name: proxy-container
+    ports:
+    - containerPort: 80
+  - env:
+    - name: ENV
+      value: dev
+    image: k8s.gcr.io/busybox
+    name: test-container
+```
+
+Please notice how:
+
+1. you still have the extra container which is patched in the base `kustomization.yaml` file
+1. the environment variable was changed to `dev`
+
+You can imagine that you can have more folders and more `kustomization.yaml` to match your clusters and environments.
+
+_Can you also edit fields without having to create a folder, a `kustomization.yaml` and the extra YAML?_
+
+Kustomize supports inline editing for some fields.
+
+Kustomize is more structured that `yq` but this convenience comes with some drawbacks.
+
+Kustomize is designed on purpose to [make it hard to change values through the command line](https://github.com/kubernetes-sigs/kustomize/blob/master/docs/eschewedFeatures.md#unstructured-edits).
+
+You have to write YAML files to change even a single value — unlike `yq`.
+
+_Is there an alternative to Kustomize and `yq` that is flexible and structured?_
+
+## Writing YAML with code
+
+You could generate YAML programatically with code.
+
+And you don't even need to start from scratch.
+
+Kubernetes has several official libraries where you can create objects such as Deployments and Pod with code.
+
+> Please note that since Kubernetes uses an OpenAPI, you can autogenerate a library for your favourite language — if there isn't one already.
+
+As an example, let's have a look at how you can use Javascript to create a Pod and programatically change it.
+
```

Comment:
First install the necessary Kubernetes JavaScript libraries:

```bash
npm init -y
npm install @kubernetes/client-node kubernetes-models
```

Reply:
This is in the other repo. does it make sense to have it here as well?
I thought I could include only the relevant code and a link to the full code.
I think the focus is not on the completeness of the setup, but rather the intent of the code.
Looking for feedback here.